src/sagemaker/estimator.py

```diff
@@ -1363,6 +1365,17 @@ def __init__(
                 authentication if they are provided; otherwise, python SDK will
                 try to use either CodeCommit credential helper or local
                 credential storage for authentication.
+            checkpoint_s3_uri (str): The S3 URI in which to persist checkpoints
```

Comment:
How would this work with https://github.com/aws/sagemaker-python-sdk/blob/152d177138f7dabe11c98938a66aeaab9acabfa7/src/sagemaker/tensorflow/estimator.py#L211

Reply:
Added an integ test to validate the training job API call has expected CheckpointConfig.

The checkpoint_path is deprecated in the tensorflow script mode:

`AttributeError: training_steps, evaluation_steps, requirements_file, checkpoint_path are deprecated in script mode. Please do not set checkpoint_path.` (https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/tensorflow/estimator.py#L338)

Reply:
How does this fare for users of the TensorFlow framework mode?

Reply:
For users of TensorFlow framework mode, passing `checkpoint_s3_uri` behaves the same as script mode. Its value get passed to EstimatorBase eventually and set the training args to CheckpointConfig and do multi-part uploads checkpointing.